core/src/main/java/org/aerogear/mobile/core/MobileCore.java

```diff
@@ -65,7 +68,11 @@ private MobileCore(final Context context, final Options options) throws Initiali
 
         // -- Setting default http layer
         if (options.httpServiceModule == null) {
-            final OkHttpServiceModule httpServiceModule = new OkHttpServiceModule();
+            OkHttpClient.Builder builder = new OkHttpClient.Builder();
+            builder.connectTimeout(10, TimeUnit.SECONDS)
+                .writeTimeout(10, TimeUnit.SECONDS)
+                .readTimeout(30, TimeUnit.SECONDS);
```

Comment:
magic numbers! 

Also I think these value should be configurable by the developers, but it probably can be done in a different PR.

Reply:
if (options.httpServiceModule == null)

Reply:
Personally I think it's not very good user experience to ask the developers to provide a whole new http implementation just to change the default timeout values.  Considering there will be more features will be added in the future to the http module (like automatically do certificate pinning for AG services), I think it's better to give some configuration options to the developers so that they can customise some aspects of the http module. 

But my main concern with this PR is the usage of magic numbers. The configuration can be dealt with in a separate PR.

Reply:
I agree with @wei-lee but we can create a new Jira/PR for it

Reply:
I agree with @wei-lee but we can create a new Jira/PR for it